src/wallet/mod.rs

```diff
@@ -49,13 +49,36 @@ use crate::{
     cli::WalletConfig,
     convert,
     messages::{self, CoinbaseBuilder, M8_BMM_REQUEST_TAG},
+    proto::mainchain,
     types::{Ctip, SidechainAck, SidechainNumber, SidechainProposal},
     validator::Validator,
 };
 use error::WalletError;
 
 pub mod error;
 
+#[derive(Debug)]
+pub struct Deposit {
+    pub sidechain_number: u8,
```

Comment:
`SidechainNumber`?

Reply:
I don't think the type wrapper for `sidechain_number` makes sense, it is already clear what it is just by the field name, adding the type wrapper just adds boilerplate without any clear benefit.